AsyncEnumerator.cs

```diff
@@ -71,6 +72,8 @@ internal void SetCanceled()
                 _isCanceled = true;
                 _enumerationException = null;
                 Interlocked.Exchange(ref _completionLock, 0);
+                if (_resumeTCS != null && !_resumeTCS.Task.IsCompleted)
+                    _resumeTCS.SetException(new AsyncEnumerationDisposedException());
```

Comment:
Keep this call here, don't move to `Finalize`.
Besides, you can make it in one line: `_resumeTCS?.TrySetException(...)`